src/modules/kafka/kafka-container.ts

```diff
@@ -43,10 +43,10 @@ export class KafkaContainer extends GenericContainer {
   private zooKeeperPort?: Port;
   private saslSslConfig?: SaslSslListenerOptions;
 
-  constructor(image = KAFKA_IMAGE) {
+  constructor(image = KAFKA_IMAGE, ports?: PortWithOptionalBinding[]) {
```

Comment:
Why add `ports` to the constructor when there's the `withExposedPorts` method?

Reply:
I figured there was a reason `withExposedPorts` was being called in the ctor here instead of in `start` like the other modules. Are you suggesting we just don't make any changes to `KafkaContainer`?

Reply:
Let's omit this change from the Kafka module for now, the ports are intertwined with the SASL configuration and the ZK. It's probably more complexity than it's worth at the moment.